frontend/src/components/order/conversationThread.tsx

```diff
@@ -0,0 +1,117 @@
+import { createMessage, MessagePublic } from "@/client";
+import { RoleTag } from "@/components/common/roleTag.tsx";
+import { getAuthorizationHeader } from "@/utils.ts";
+import { Box, Button, Checkbox, Flex, Link, Textarea } from "@chakra-ui/react";
+import { Link as RouterLink, useRouter } from "@tanstack/react-router";
+import { useEffect, useRef, useState } from "react";
+import { useAuth } from "react-oidc-context";
+
+interface Props {
+  messages: MessagePublic[];
+  orderId: number;
+}
+
+export function ConversationThread({ messages, orderId }: Props) {
+  const auth = useAuth();
+  const router = useRouter();
+
+  const [messageValue, setMessageValue] = useState<string>("");
+
+  const messageBoxRef = useRef<HTMLDivElement>(null);
+
+  const scrollToBottom = () => {
+    if (messageBoxRef.current) {
+      messageBoxRef.current.scrollTop = messageBoxRef.current.scrollHeight;
+    }
+  };
+
+  useEffect(() => {
+    scrollToBottom();
+  }, [messages]);
+
+  const handleSendMessage = async () => {
+    try {
+      await createMessage({
+        headers: { ...getAuthorizationHeader(auth) },
+        body: {
+          content: messageValue,
+          order_id: orderId,
+        },
+      });
+      // Invalidate the current route to reload messages
+      router.invalidate();
+      setMessageValue("");
+    } catch (error) {
+      // TODO: Better error handling
+      console.error("Failed to send message:", error);
+    }
+  };
+
+  return (
+    <>
+      <Box pb="2" fontSize="lg" fontWeight="bold">
+        Order timeline
+      </Box>
+      <Box p={6} boxShadow="md" borderRadius="md" bg="blue.50">
+        <Flex
+          direction="column"
+          gap={4}
+          maxH="400px"
+          overflowY="auto"
+          ref={messageBoxRef}
+        >
+          {messages && messages.length > 0 ? (
+            messages.map((message, index) => (
```

Comment:
I guess that having index here can cause unnecessary rerenders when messages are updated